### PR TITLE
[Bugfix] Fixed the archiving issue introduced by using @testable import in demo app

### PR DESCRIPTION
**Pull request by kevinyulu**
## Purpose

* Fixed the archiving issue introduced by using @testable import in demo app

Issue:
Screen Shot 2022-11-23 at 2 17 39 PM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallConfigurationMocking.swift [hunk 1]
```diff
@@ -32,3 +33,5 @@ enum CompositeCallTypeMocking {
     case groupCall
     case teamsMeeting
 }
+#else
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallConfigurationMocking.swift:36]: No need for an `#else` if we don't have something in the clause

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallingSDKEventsHandlerMocking.swift [hunk 2]
```diff
@@ -55,3 +56,6 @@ class CallingSDKEventsHandlerMocking: CallingSDKEventsHandler {
         }
     }
 }
+
+#else
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallingSDKEventsHandlerMocking.swift:60]: Same here.

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/UIKitDemoViewController.swift [hunk 3]
```diff
@@ -161,11 +165,14 @@ class UIKitDemoViewController: UIViewController {
...
             : Theming(envConfigSubject: envConfigSubject),
             localization: localizationConfig)
-
+        #if DEBUG
         let callComposite = envConfigSubject.useMockCallingSDKHandler ?
             CallComposite(withOptions: callCompositeOptions,
                           callingSDKWrapperProtocol: UITestCallingSDKWrapper())
             : CallComposite(withOptions: callCompositeOptions)
+        #else
+        let callComposite = CallComposite(withOptions: callCompositeOptions)
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/UIKitDemoViewController.swift:174]: Should the control to enable the mock calling SDK also be disabled in this case?

**Review by emlynmac — COMMENTED** [review]
Couple of minor things and a question about the UI to use the mock calling SDK

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/SwiftUIDemoView.swift [hunk 4]
```diff
@@ -163,10 +166,14 @@ extension SwiftUIDemoView {
             : Theming(envConfigSubject: envConfigSubject),
             localization: localizationConfig)
         let useMockCallingSDKHandler = envConfigSubject.useMockCallingSDKHandler
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/SwiftUIDemoView.swift:168]: `let useMockCallingSDKHandler = envConfigSubject.useMockCallingSDKHandler`
can be moved under `#if DEBUG`

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/UITestCallingSDKWrapper.swift [hunk 5]
```diff
@@ -165,10 +166,8 @@ extension UITestCallingSDKWrapper {
...
     private func makeCallClientMocking() -> CallClientMocking {
         let clientOptionsMocking = CallClientOptionsMocking()
-        let appendingTag = self.callConfigurationMocking.diagnosticConfig.tags
-        let diagnostics = clientOptionsMocking.diagnostics ?? CallDiagnosticsOptionsMocking()
-        diagnostics.tags.append(contentsOf: appendingTag)
-        clientOptionsMocking.diagnostics = diagnostics
         return CallClientMocking(options: clientOptionsMocking)
     }
 }
+#else
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/UITestCallingSDKWrapper.swift:172]: Remove this #else?

**Review by emlynmac — COMMENTED** [review]
One minor issue, otherwise LGTM

**Review by emlynmac — APPROVED** [review]
LGTM

**Review by jimchou-dev — APPROVED** [review]
LGTM

**Pull request merged**